### PR TITLE
Configurable lifetime for static files

### LINKED ISSUE #7: Make cache lifetime for static files configurable
The cache lifetime sent in an HTTP response as the "expires" header should be configurable via an environment variable.

### PR DESCRIPTION
**Pull request by robertlemke**
This change introduces a new environment variable
NGINX_STATIC_FILES_LIFETIME which allows for configuration of the
cache lifetime controlled through the HTTP response's "expires"
header. The default expiration time is now 30 days.

Apart from Flow's static resources, additional files in the top
directory are now covered (favicon.ico, apple-touch-icon.* etc).

Resolves #7

### CODE REVIEW

**Comment by robertlemke** [comment]
Ah, this also fixes the issue in Local Beach where requests to non-existing static files would end up triggering a request in Flow / Neos. Now missing static files just end up in a 404 directly.

File: root-files/opt/flownative/lib/nginx-legacy.sh [hunk 1]
```diff
@@ -184,7 +191,7 @@ EOM
     # pass persistent resource requests to GCS
     location ~* "^${BEACH_PERSISTENT_RESOURCES_BASE_PATH}([a-f0-9]{40})/" {
         resolver 8.8.8.8;
-        expires 3600;
+        expires ${NGINX_STATIC_FILES_LIFETIME};
```
Comment (kdambekalns) [inline root-files/opt/flownative/lib/nginx-legacy.sh:194]: I'd not expect that `_STATIC_` variable to have an effect on `_Persistent_` resources…

File: root-files/opt/flownative/lib/nginx-legacy.sh [hunk 2]
```diff
@@ -193,9 +200,9 @@ EOM
 EOM
     elif [ -n "${BEACH_PERSISTENT_RESOURCES_FALLBACK_BASE_URI}" ]; then
         cat >>"${NGINX_CONF_PATH}/sites-enabled/site.conf" <<-EOM
-    location ~* ^/_Resources/Persistent/(.*)$ {
+    location ~* "^${BEACH_PERSISTENT_RESOURCES_BASE_PATH}(.*)$ {
         access_log off;
-        expires 3600;
+        expires ${NGINX_STATIC_FILES_LIFETIME};
```
Comment (kdambekalns) [inline root-files/opt/flownative/lib/nginx-legacy.sh:205]: See above, same here on static vs. persistent.

File: root-files/opt/flownative/lib/nginx-legacy.sh [hunk 3]
```diff
@@ -209,7 +216,16 @@ EOM
         proxy_pass \$assetUri;
     }
 EOM
-
+    else
+        cat >>"${NGINX_CONF_PATH}/sites-enabled/site.conf" <<-EOM
+    location ~* ^/_Resources/Persistent/(.*)$ {
+        access_log off;
+        expires ${NGINX_STATIC_FILES_LIFETIME};
```
Comment (kdambekalns) [inline root-files/opt/flownative/lib/nginx-legacy.sh:223]: … and here, too.

File: root-files/opt/flownative/lib/nginx-legacy.sh [hunk 4]
```diff
@@ -193,9 +200,9 @@ EOM
 EOM
     elif [ -n "${BEACH_PERSISTENT_RESOURCES_FALLBACK_BASE_URI}" ]; then
         cat >>"${NGINX_CONF_PATH}/sites-enabled/site.conf" <<-EOM
-    location ~* ^/_Resources/Persistent/(.*)$ {
+    location ~* "^${BEACH_PERSISTENT_RESOURCES_BASE_PATH}(.*)$ {
```
Comment (kdambekalns) [inline root-files/opt/flownative/lib/nginx-legacy.sh:203]: Is that always the same value as before?

File: root-files/opt/flownative/lib/nginx-legacy.sh [hunk 5]
```diff
@@ -209,7 +216,16 @@ EOM
         proxy_pass \$assetUri;
     }
 EOM
-
+    else
+        cat >>"${NGINX_CONF_PATH}/sites-enabled/site.conf" <<-EOM
+    location ~* ^/_Resources/Persistent/(.*)$ {
```
Comment (kdambekalns) [inline root-files/opt/flownative/lib/nginx-legacy.sh:221]: Compared to above, why is this still hardcoded and not using a variable?

File: root-files/opt/flownative/lib/nginx-legacy.sh (hunk 4 above)
Comment (robertlemke) [reply root-files/opt/flownative/lib/nginx-legacy.sh:203]: If someone set the `BEACH_PERSISTENT_RESOURCES_BASE_PATH` to something else for a fallback-setup, this wouldn't have had any effect. So I guess we can dare to apply this change.

File: root-files/opt/flownative/lib/nginx-legacy.sh (hunk 5 above)
Comment (robertlemke) [reply root-files/opt/flownative/lib/nginx-legacy.sh:221]: This else case handles a setup without cloud storage, but storing persistent resources in a local file system. I mean, we possibly _can_ support other directories, for that, but does it make sense?

File: root-files/opt/flownative/lib/nginx-legacy.sh (hunk 1 above)
Comment (robertlemke) [reply root-files/opt/flownative/lib/nginx-legacy.sh:194]: The "static" refers to files in the local file system, from the Nginx point of view (versus dynamic data, provided by PHP-FPM for example).

There's also a variable called "NGINX_STATIC_ROOT" which uses "static" for a similar purpose.

**Comment by robertlemke** [comment]
I will merge and test this in a beta setup and don't communicate the variable names to customers yet, to see if the cache lifetime configuration works in production as expected.

**Pull request merged**